File: Semestre_10/Electro/Tareas/Tarea3/punto_1.py
import numpy as np
import matplotlib.pyplot as plt

class Element():
    def __init__(self, name, A, B) -> None:
        self.name = name
        self.A = A
        self.B = B

    def calculate_n(self):
        space = np.linspace(0.4, 0.7, 100)
        return (space, self.A + (self.B / np.pow(space, 2)))

if __name__=="__main__":
    elements = [
        Element("Silica", 1.4580, 0.00354),
        Element("BK7", 1.5046, 0.00420)
    ]
    plt.figure(figsize=(10, 6))
    for element in elements:
        x, y = element.calculate_n()
        plt.plot(x, y, label=element.name)

    plt.xlabel('Longitud de onda (μm)')
    plt.ylabel('Índice de refracción n(λ)')
    plt.title('Índice de refracción según la ecuación de Cauchy')
    plt.legend()
    plt.grid(True)
    plt.show()

File: Semestre_10/Electro/Tareas/Tarea3/punto_1.R
library(ggplot2)

elements <- read.csv("./punto_1.csv")

calculate_n <- function(element) {
  space <- seq(0.4, 0.7, length.out = 100)
  n_values <- element$A + (element$B / space^2)
  data.frame(wavelength = space, n = n_values, element = element$name)
}


plot_data <- calculate_n(elements)

plot_data |>
ggplot(aes(x = wavelength, y = n, color = element)) +
  geom_line() +
  labs(
    x = "Longitud de onda (μm)",
    y = "Índice de refracción n(λ)",
    title = "Índice de refracción según la ecuación de Cauchy"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank())

ggsave("punto_1.png")
